src/main.lua

```diff
@@ -176,10 +178,19 @@ function love.update(dt)
       else
         roadShift = "center"
       end
-      road:update(p:getTrajectory(), roadShift)
+      x, y, angle = road:pushFrontier(p:getTrajectory(), roadShift)
+      newSegments[i] = {}
+      newSegments[i].x = x
+      newSegments[i].y = y
+      newSegments[i].angle = angle
+      i = i + 1
       distance = love.physics.getDistance(p.fixture, road.frontier.main.fixture)
     end
-
+    
+    if table.getn(newSegments) > 0 then
```

Comment:
there is also a handy `#` operator in lua to give you the length of a table, e.g., `#newSegments`.